graylog2-server/src/main/java/org/graylog2/indexer/IndexSet.java

```diff
@@ -170,6 +170,25 @@ public interface IndexSet extends Comparable<IndexSet> {
      */
     IndexSetConfig getConfig();
 
+    default String getNthIndexBeforeActiveIndexSet(final int n) {
```

Comment:
Could we return optional instead of nulls here?

Reply:
All index-getting methods in IndexSet class return plain String, not Optional<String>. Do you believe that it is not inconsistent to use different approaches? If we believe Optionals would be better, shouldn't we have a separate task to change it everywhere in this class, to be consistent?